# SPDX-License-Identifier: BSD-2-Clause
# Origin: EmberBSD contributors (AI-assisted).
# Preserve the ICU default while allowing an explicit package variant.

--- a/devel/boost-libs/Makefile
+++ b/devel/boost-libs/Makefile
@@ -1,7 +1,7 @@
 # $NetBSD: Makefile,v 1.127 2026/05/15 09:13:49 adam Exp $
 
 BOOST_PACKAGE=		libs
-PKGREVISION=		1
+PKGREVISION=		2
 BOOST_COMMENT=		(binary libraries)
 BOOST_CONFIG=		installed
 BOOST_INSTALL_LIBS=	yes
@@ -66,7 +66,6 @@
 
 CONFIGURE_ENV+=	BZIP2_INCLUDE=${BUILDLINK_PREFIX.bzip2}/include
 CONFIGURE_ENV+=	BZIP2_LIBRARY_PATH=${BUILDLINK_PREFIX.bzip2}/lib
-CONFIGURE_ENV+=	ICU_PATH=${BUILDLINK_PREFIX.icu}
 CONFIGURE_ENV+=	ZLIB_INCLUDE=${BUILDLINK_PREFIX.zlib}/include
 CONFIGURE_ENV+=	ZLIB_LIBRARY_PATH=${BUILDLINK_PREFIX.zlib}/lib
 
@@ -94,6 +93,16 @@
 .include "../../converters/libiconv/buildlink3.mk"
 .include "../../devel/boost-headers/buildlink3.mk"
 .include "../../devel/zlib/buildlink3.mk"
+PKG_OPTIONS_VAR= PKG_OPTIONS.boost-libs
+PKG_SUPPORTED_OPTIONS= icu
+PKG_SUGGESTED_OPTIONS= icu
+.include "../../mk/bsd.options.mk"
+.if !empty(PKG_OPTIONS:Micu)
+CONFIGURE_ENV+= ICU_PATH=${BUILDLINK_PREFIX.icu}
 .include "../../textproc/icu/buildlink3.mk"
+.else
+# Regex and Locale have independent upstream ICU selectors.
+BJAM_ARGS+= --disable-icu boost.locale.icu=off
+.endif
 .include "../../mk/pthread.buildlink3.mk"
 .include "../../mk/bsd.pkg.mk"
